numpy/core/src/multiarray/einsum_sumprod.h

```diff
@@ -0,0 +1,12 @@
+#ifndef _NPY_MULTIARRAY_EINSUM_SUMPROD_H
+#define _NPY_MULTIARRAY_EINSUM_SUMPROD_H
+
+#include <numpy/npy_common.h>
+
+typedef void (*sum_of_products_fn)(int, char **, npy_intp const*, npy_intp);
+
+NPY_VISIBILITY_HIDDEN sum_of_products_fn
+get_sum_of_products_function(int nop, int type_num,
+                             npy_intp itemsize, npy_intp const *fixed_strides);
```

Comment:
```suggestion
/**
 * forward declarations according to configuration statements in
 * the dispatch-able source 'einsum.dispatch.c.src',
 * see 'npy_cpu_dispatch.h' for more clarification.
 */
#ifndef NPY_DISABLE_OPTIMIZATION
    // auto-generated config header required by NPY_CPU_DISPATCH_DECLARE
    #include "einsum.dispatch.h"
#endif
NPY_CPU_DISPATCH_DECLARE(NPY_VISIBILITY_HIDDEN sum_of_products_fn simd_sum_of_products_function,
    (int nop, int type_num, npy_intp itemsize, npy_intp const *fixed_strides))
```

Reply:
`get_sum_of_products_function` reserved for runtime dispatching call, `simd_sum_of_products_function` is used instead